moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp

```diff
@@ -105,53 +120,71 @@ void RobotModelLoader::configure(const Options& opt)
 
-        double max_position;
-        if (nh.getParam(prefix + "max_position", max_position))
+        std::string param_name;
+        try
         {
-          if (canSpecifyPosition(joint_model, joint_id))
+          param_name = prefix + "max_position";
+          double max_position;
+          if (node_->get_parameter(param_name, max_position))
           {
-            joint_limit[joint_id].has_position_limits = true;
-            joint_limit[joint_id].max_position = max_position;
+            if (canSpecifyPosition(jmodel, j))
+            {
+              jlim[j].has_position_limits = true;
+              jlim[j].max_position = max_position;
+            }
           }
-        }
-        double min_position;
-        if (nh.getParam(prefix + "min_position", min_position))
-        {
-          if (canSpecifyPosition(joint_model, joint_id))
+
+          param_name = prefix + "min_position";
+          double min_position;
+          if (node_->get_parameter(param_name, min_position))
           {
-            joint_limit[joint_id].has_position_limits = true;
-            joint_limit[joint_id].min_position = min_position;
+            if (canSpecifyPosition(jmodel, j))
+            {
+              jlim[j].has_position_limits = true;
+              jlim[j].min_position = min_position;
+            }
           }
-        }
-        double max_velocity;
-        if (nh.getParam(prefix + "max_velocity", max_velocity))
-        {
-          joint_limit[joint_id].has_velocity_limits = true;
-          joint_limit[joint_id].max_velocity = max_velocity;
-        }
-        bool has_vel_limits;
-        if (nh.getParam(prefix + "has_velocity_limits", has_vel_limits))
-          joint_limit[joint_id].has_velocity_limits = has_vel_limits;
 
-        double max_acc;
-        if (nh.getParam(prefix + "max_acceleration", max_acc))
+          param_name = prefix + "max_velocity";
+          double max_velocity;
+          if (node_->get_parameter(param_name, max_velocity))
+          {
+            jlim[j].has_velocity_limits = true;
+            jlim[j].max_velocity = max_velocity;
+          }
+
+          param_name = prefix + "has_velocity_limits";
+          bool has_vel_limits;
+          if (node_->get_parameter(param_name, has_vel_limits))
+          {
+            jlim[j].has_velocity_limits = has_vel_limits;
+          }
+
+          param_name = prefix + "max_acceleration";
+          double max_acc;
+          if (node_->get_parameter(param_name, max_acc))
+          {
+            jlim[j].has_acceleration_limits = true;
+            jlim[j].max_acceleration = max_acc;
+          }
+
+          param_name = prefix + "has_acceleration_limits";
+          bool has_acc_limits;
+          if (node_->get_parameter(param_name, has_acc_limits))
+            jlim[j].has_acceleration_limits = has_acc_limits;
+        }
+        catch (const rclcpp::ParameterTypeException& e)
```

Comment:
We should probably add this pattern to as parameter best practices to the migration guide https://github.com/ros-planning/moveit2/blob/b96a0259eb8d35364ab7d52f26fd50cdd45ee513/doc/MIGRATION_GUIDE.md